### PR TITLE
feat: hardhat custom ethers functionalities

### PR DESCRIPTION
**Pull request by rodolfopietro97**
# Description

Implement missing hardhat ethers functionalities

### CODE REVIEW

**Review by fabiorigam — CHANGES_REQUESTED** [review]
Please increase coverage

File: apps/sdk-hardhat-integration/scripts/deploy.ts [hunk 1]
```diff
@@ -1,29 +1,20 @@
...
 
-    await vechainHelloWorldWithNonEmptyConstructor.waitForDeployment();
+    console.log('factory', JSON.stringify(vechainHelloWorldFactory));
 
-    console.log(
-        `VechainHelloWorld deployed to ${JSON.stringify(vechainHelloWorld)}`
-    );
+    const txResponse = await vechainHelloWorldFactory.deploy();
+
+    console.log('txResponse', JSON.stringify(txResponse));
```
Comment (fabiorigam) [inline apps/sdk-hardhat-integration/scripts/deploy.ts:15]: DO we need all these console.log? Let's just keep one like "Contract x deployed to ..."

File: packages/ethers-adapter/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# @vechain/sdk-hardhat-plugin
+
+Welcome to the **hardhat-plugin package** of the vechain SDK!
+
+## Introduction
+
```
Comment (fabiorigam) [inline packages/ethers-adapter/README.md:6]: This is a copy and paste, please update the text

File: packages/ethers-adapter/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "@vechain/sdk-ethers-adapter",
+  "version": "1.0.0-beta.2",
+  "description": "This module allows to create smart contracts and interact with them using the vechain SDK and Hardhat",
```
Comment (fabiorigam) [inline packages/ethers-adapter/package.json:4]: same here

File: packages/ethers-adapter/package.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  "author": "vechain Foundation",
+  "license": "MIT",
+  "keywords": [
+    "vechain",
+    "hardhat",
+    "ethers",
```
Comment (fabiorigam) [inline packages/ethers-adapter/package.json:10]: update keywords

File: packages/ethers-adapter/src/adapters/contract-adapter.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { type vechain_sdk_core_ethers } from '@vechain/sdk-core';
+import { type HardhatVechainProvider } from '@vechain/sdk-provider';
+import { helpers } from './helpers';
+
+/**
+ * Contract adapter for the vechain hardhat plugin
+ *
+ * @param contract - The contract to adapt to the vechain network
+ * @param hardhatVechainProvider - The hardhat vechain provider
+ */
```
Comment (fabiorigam) [inline packages/ethers-adapter/src/adapters/contract-adapter.ts:10]: missing @returns

File: packages/ethers-adapter/src/adapters/factory-adapter.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
+import type { HardhatVechainProvider } from '@vechain/sdk-provider';
+import { vechain_sdk_core_ethers } from '@vechain/sdk-core/dist';
+
+/**
+ * Factory adapter for the vechain hardhat plugin
+ *
+ * @param contractFactory - The contract factory to adapt to the vechain network
+ * @param hardhatVechainProvider - The hardhat vechain provider
+ */
```
Comment (fabiorigam) [inline packages/ethers-adapter/src/adapters/factory-adapter.ts:9]: missing @returns

File: packages/ethers-adapter/src/adapters/helpers.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import type { HardhatVechainProvider } from '@vechain/sdk-provider';
+
+/**
+ * Get the contract address from a transaction id
+ * @param txId - The contract deployment transaction id
+ * @param hardhatVechainProvider - The hardhat vechain provider
+ */
```
Comment (fabiorigam) [inline packages/ethers-adapter/src/adapters/helpers.ts:7]: same

**Review by fabiorigam — CHANGES_REQUESTED** [review]

File: packages/ethers-adapter/README.md (hunk 2 above)
Comment (Valazan) [reply packages/ethers-adapter/README.md:6]: updated

**Review by fabiorigam — APPROVED** [review]

**Pull request merged**